Detectors/TRD/base/src/PadPlane.cxx

```diff
@@ -158,3 +162,59 @@ void PadPlane::setNrows(int n)
   }
   mNrows = n;
 };
+
+double PadPlane::getPadRow(double z) const
+{
+  double lengthCorr = mLengthIPad / mLengthOPad;
+
+  // calculate position based on inner pad length
+  double padrow = -z / mLengthIPad + mNrows / 2;
```

Comment:
In the same direction, create `mInverseLengthIPad` as a class member and set it when `mLengthIPad` is defined to avoid repetitive division.